### PR TITLE
fix: wait for kubeadmin password secret to be populated

### PR DESCRIPTION
**Pull request by rhopp**

### CODE REVIEW

**Comment by rhopp** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e

**Comment by rhopp** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e

**Comment by rhopp** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e

**Comment by rhopp** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e

**Comment by rhopp** [comment]
/pj-rehearse pull-ci-redhat-appstudio-e2e-tests-main-redhat-appstudio-e2e

File: ci-operator/step-registry/hypershift/aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml [hunk 1]
```diff
@@ -59,10 +59,18 @@ chain:
...
-
+      timeout --foreground 5m bash <<- "EOF"
+        while [[ -z "$(oc get --namespace=${HYPERSHIFT_NAMESPACE} hostedcluster ${CLUSTER_NAME} -o jsonpath='{.status.kubeadminPassword.name}')" ]]; do
+          echo "Waiting for kubeadmin password to be populated"
+          sleep 5
+        done
+        KUBEADMIN_SECRET_NAME=$(oc get hostedcluster --namespace=${HYPERSHIFT_NAMESPACE} ${CLUSTER_NAME} -o jsonpath='{.status.kubeadminPassword.name}')
+        while ! KUBEADMIN_SECRET_ENCODED=$(oc get secret ${KUBEADMIN_SECRET_NAME} -n${HYPERSHIFT_NAMESPACE}  --template={{.data.password}}); do
+          echo "Kubeadmin secret does not exist. Waiting for 5 seconds"
+          sleep 5
```
Comment (enxebre) [inline ci-operator/step-registry/hypershift/aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml:70]: why is this sleep needed given we already check status above this loop shouldn't be needed

File: ci-operator/step-registry/hypershift/aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml [hunk 2]
```diff
@@ -59,10 +59,18 @@ chain:
 
       # Data for cluster bot.
       # The kubeadmin-password secret is reconciled only after the kas is available so giving a few seconds to avoid racing and move forward otherwise. 
-      sleep 15
-      echo "Getting ${CLUSTER_NAME}-kubeadmin-password from namespace ${HYPERSHIFT_NAMESPACE}"
-      oc get secret -n${HYPERSHIFT_NAMESPACE} ${CLUSTER_NAME}-kubeadmin-password --template={{.data.password}} | base64 -d > ${SHARED_DIR}/kubeadmin-password
-
+      timeout --foreground 5m bash <<- "EOF"
```
Comment (enxebre) [inline ci-operator/step-registry/hypershift/aws/setup-nested-management-cluster/hypershift-aws-setup-nested-management-cluster-chain.yaml:62]: if we happen to have a bug on this logic or in the core logic that creates the secret we don't want to block jobs for 5m here.

This was only used so cluster-bot could print the value.

**Comment by rhopp** [comment]
We switched to https://github.com/openshift/release/blob/master/ci-operator/step-registry/redhat-appstudio/install-rhtap/redhat-appstudio-install-rhtap-commands.sh#L48-L68 which has the wait logic implemented and don't see this problem anymore. Closing this PR as not needed anymore.

**Pull request closed without merging**